substratevm/src/com.oracle.svm.jfr/src/com/oracle/svm/jfr/JfrEventSubstitution.java

```diff
@@ -103,7 +103,15 @@ private static void changeWriterResetMethod(ResolvedJavaType eventWriterType) {
     private static void setPublicModifier(ResolvedJavaMethod m) {
         try {
             Class<?> method = m.getClass();
-            Method metaspaceMethodM = method.getDeclaredMethod("getMetaspaceMethod");
+            Method metaspaceMethodM = null;
+            try {
+                metaspaceMethodM = method.getDeclaredMethod("getMetaspaceMethod");
+            } catch (NoSuchMethodException e) {
+                // Fall back for OpenJDK 11u-based builds:
+                //  getMetaspaceMethod() got introduced into OpenJDK 13 with JDK-8220623.
+                //  Plain OpenJDK 11u-based code won't have it. Use getMetaspacePointer() instead.
+                metaspaceMethodM = method.getDeclaredMethod("getMetaspacePointer");
```

Comment:
`getMetaspacePointer` should work on all JDK versions, so please adapt that code so that it always uses `getMetaspacePointer` (i.e., remove `method.getDeclaredMethod("getMetaspaceMethod");`). This will also make the comment that you added obsolete.

Reply:
Even better ;-)